frontend/javascripts/oxalis/model/sagas/isosurface_saga.js

```diff
@@ -369,6 +358,164 @@ function* maybeLoadIsosurface(
+  modifiedCells.clear();
+
+  const segmentationLayer = Model.getVisibleSegmentationLayer();
+  if (!segmentationLayer) {
+    return;
+  }
+  // First create an array containing information about all loaded isosurfaces as the map is manipulated within the loop.
+  for (const [cellId, threeDMap] of Array.from(
+    isosurfacesMapByLayer[segmentationLayer.name].entries(),
+  )) {
+    if (!currentlyModifiedCells.has(cellId)) {
+      continue;
+    }
+
+    yield* call(_refreshIsosurfaceWithMap, cellId, threeDMap, segmentationLayer.name);
+  }
+}
+
+function* refreshIsosurface(action: RefreshIsosurfaceAction): Saga<void> {
+  const { cellId, layerName } = action;
+
+  const threeDMap = isosurfacesMapByLayer[action.layerName].get(cellId);
+  if (threeDMap == null) return;
+
+  yield* call(_refreshIsosurfaceWithMap, cellId, threeDMap, layerName);
+}
+
+function* _refreshIsosurfaceWithMap(
+  cellId: number,
+  threeDMap: ThreeDMap<boolean>,
+  layerName: string,
+): Saga<void> {
+  const isosurfacePositions = threeDMap.entries().filter(([value, _position]) => value);
+  if (isosurfacePositions.length === 0) {
+    return;
+  }
+  yield* put(startedLoadingIsosurfaceAction(layerName, cellId));
+  // Remove isosurface from cache.
+  yield* call(removeIsosurface, removeIsosurfaceAction(layerName, cellId), false);
+  // The isosurface should only be removed once after re-fetching the isosurface first position.
+  let shouldBeRemoved = true;
+  for (const [, position] of isosurfacePositions) {
+    // Reload the isosurface at the given position if it isn't already loaded there.
+    // This is done to ensure that every voxel of the isosurface is reloaded.
+    yield* call(ensureSuitableIsosurface, position, cellId, shouldBeRemoved);
+    shouldBeRemoved = false;
+  }
+  yield* put(finishedLoadingIsosurfaceAction(layerName, cellId));
+}
+
+/*
+ *
+ * Precomputed Meshes
+ *
+ */
+
+function* loadPrecomputedMesh(action: LoadPrecomputedMeshAction) {
+  const { cellId, seedPosition, meshFileName, layerName } = action;
+  const layer = yield* select(state =>
+    layerName != null
+      ? getSegmentationLayerByName(state.dataset, layerName)
+      : getVisibleSegmentationLayer(state),
+  );
+  if (layer == null) return;
+
+  // If a REMOVE_ISOSURFACE action is dispatched and consumed
+  // here before loadPrecomputedMeshForSegmentId is finished, the latter saga
+  // should be canceled automatically to avoid populating mesh data even though
+  // the mesh was removed. This is accomplished by redux-saga's race effect.
+  yield* race({
+    loadPrecomputedMeshForSegmentId: _call(
+      loadPrecomputedMeshForSegmentId,
+      cellId,
+      seedPosition,
+      meshFileName,
+      layer,
+    ),
+    cancel: _take(
+      otherAction =>
+        otherAction.type === "REMOVE_ISOSURFACE" &&
+        otherAction.cellId === cellId &&
+        otherAction.layerName === layerName,
+    ),
+  });
+}
+
+function* loadPrecomputedMeshForSegmentId(
```

Comment:
This function was moved from `segments_view_helper.js` and converted to a saga. Because of the race- and cancel-ability of sagas, the numerous checks whether the mesh was deleted in the mean time could be removed.